test/common/performance/aggregator/aggregator.go

```diff
@@ -57,20 +61,32 @@ type aggregatorExecutor struct {
 	listener net.Listener
 	server   *grpc.Server
 
-	makoTags []string
+	makoTags      []string
+	expectRecords uint
+
+	benchmarkKey  string
+	benchmarkName string
 }
 
-func newAggregatorExecutor(lis net.Listener, makoTags []string) testExecutor {
-	executor := &aggregatorExecutor{
-		listener:             lis,
+func NewAggregator(benchmarkKey string, benchmarkName string, listenAddr string, expectRecords uint, makoTags []string) (common.Executor, error) {
+	l, err := net.Listen("tcp", listenAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create listener: %v", err)
+	}
+
+	executor := Aggregator{
+		listener:             l,
 		notifyEventsReceived: make(chan struct{}),
 		makoTags:             makoTags,
+		expectRecords:        expectRecords,
+		benchmarkKey:         benchmarkKey,
+		benchmarkName:        benchmarkName,
 	}
 
 	// --- Create GRPC server
 
 	s := grpc.NewServer(grpc.MaxRecvMsgSize(maxRcvMsgSize))
-	pb.RegisterEventsRecorderServer(s, executor)
+	pb.RegisterEventsRecorderServer(s, &executor)
```

Comment:
Since you use references everywhere, you could simply create a pointer to a struct above.